book/visual-effects.md

```diff
@@ -1493,43 +1436,40 @@ fits perfectly. In code, destination-in looks like this:
 class Pixel:
     def destination_in(self, source):
         self.a = self.a * source.a
-        if self.a == 0: return self
-        self.r = (self.r * self.a * source.a) / self.a
```

Comment:
Why does all of this logic go?

Reply:
It cancels out.

you get self.r * self.a * source.a / new_a, where new_a is self.a * source.a, cancel out and it’s just self.r = self.r